test/test_log.js

```diff
@@ -0,0 +1,113 @@
+var SplunkLogger = require("../index").Logger;
+var assert = require("assert");
+var sinon = require("sinon");
```

Comment:
It's a nice way to mock a function, so instead of doing a full integration test to Splunk, I could run it without ever making the call to Splunk and see that the functions are called with the correct output.  

It also could be used to mock the clock, so if for instance you wanted to test that the flush function is called every 1 second you can tick 999 ms and see no call and at 1000 ms the function is called.  Also that technique could be used to speed up the integration tests to immediately fast for 1 second so the test can complete in a few ms versus waiting for the function to be triggered.